src/item.cpp

```diff
@@ -3884,20 +3902,42 @@ int item::gun_pierce( bool with_ammo ) const
     return ret;
 }
 
-int item::gun_recoil( bool with_ammo ) const
+int item::gun_recoil( const player &p, bool bipod ) const
 {
-    if( !is_gun() ) {
+    if( !is_gun() || ( ammo_required() && !ammo_remaining() ) ) {
         return 0;
     }
-    int ret = type->gun->recoil;
-    if( with_ammo && ammo_data() ) {
-        ret += ammo_data()->ammo->recoil;
-    }
+
+    ///\EFFECT_STR improves the handling of heavier weapons
+    // we consider only base weight to avoid exploits
+    double wt = std::min( type->weight, p.str_cur * 250 ) / 250.0;
+
+    double handling = type->gun->handling;
     for( const auto mod : gunmods() ) {
-        ret += mod->type->gunmod->recoil;
+        if( bipod || !mod->has_flag( "BIPOD" ) ) {
+            handling += mod->type->gunmod->handling;
+        }
+    }
+
+    // rescale from JSON units which are intentionally specified as integral values
+    handling /= 10;
+
+    // algorithm is biased so heavier weapons benefit more from improved handling
+    handling = pow( wt, 0.8 ) * pow( handling, 1.2 );
+
+    int qty = type->gun->recoil;
+    if( ammo_data() ) {
+        qty += ammo_data()->ammo->recoil;
+    }
+
+    // handling could be either a bonus or penalty dependent upon installed mods
+    if( handling > 0.0 ) {
```

Comment:
Should be centered around 1, not 0. Currently 0 handling is better than 0.5 handling and -0.1 handling is better than 1 handling.